File: src/lecture2/linear_insertion/Main.java
public class Main {
  public static void main(String[] args) {
    int[] arr = new int[]{1, 2, 3, 4, 5, 6, 0, 0, 0};
    var main = new Main();
    main.insert(arr, 6, 2, 22);
    for (int x : arr) {
      System.out.println(x);
    }
  }

  public void insert(int[] arr, int size, int pos, int value) {
    if (arr.length == size) {
      System.out.println("no space left");
      return;
    }
    if (pos >= arr.length || pos < 0) {
      System.out.println("invalid index given");
      return;
    }

    for (int i = size - 1; i >= pos; i--) {
      arr[i + 1] = arr[i];
    }
    arr[pos] = value;
  }
}

File: src/lecture2/linear_insertion/main.py
import array as arr

def insert(arr, size, pos, data):
  for i in reversed (range(pos, size)):
    arr[i + 1] = arr[i]
  arr[pos] = data

if __name__=="__main__":
  linear = arr.array("i", [11, 22, 33, 44, 55, 66, 0, 0, 0])
  insert(linear, 6, 2, 99)
  for i in linear:
    print(i)